### PR TITLE
feat(api): add Nanomodule REST endpoints and Tests

### LINKED ISSUE #426: [Moodle-Integration] REST API for querying for Nanomodules
Create an REST API to search for Nanomodules (lessons) as part of the `lesson.router`. The endpoints should be analogue to the ones of the `course.router`.
- [ ] Search for lessons (by name / author, with pagination)
- [ ] Details os a lesson for a given slug
- [ ] Progress (1 or 0), if the lesson is done by a set of given students, if the caller is also the author of the lesson.

### PR DESCRIPTION
**Pull request by AsadullahBajwa**
## Add REST API endpoints for lessons(nanomodules)

Implements three new lesson endpoints.

**Endpoints:**
* `GET /lessons` - List and filter lessons with pagination
* `GET /lessons/{slug}` - Get lesson details by slug
* `GET /lessons/{slug}/progress` - Query student completion status

**Features:**
* Returns binary completion (0/1) for specified students
* Only shows data for valid users in system
* Restricted to lesson authors only
* Handles edge cases (non-existent lessons, unauthorized access)

Includes comprehensive test coverage for all endpoints.

### CODE REVIEW

File: libs/data-access/api/src/lib/trpc/routers/lesson.router.ts [hunk 1]
```diff
@@ -174,6 +174,218 @@ export const lessonRouter = t.router({
 					});
 				}
 			}
+		}),
+	//
+	listAvailableLessons: authProcedure
```
Comment (spark-sse) [inline libs/data-access/api/src/lib/trpc/routers/lesson.router.ts:179]: this router feels like it is just an extension to the existing findLesson with pagination?

Refactoring the existing one would propably more fitting

File: libs/data-access/api/src/lib/trpc/routers/lesson.router.ts [hunk 2]
```diff
@@ -174,6 +174,218 @@ export const lessonRouter = t.router({
...
+					title: lesson.title,
+					slug: lesson.slug
+				})),
+				pageSize: pageSize,
+				page: input.page,
+				totalCount: count
+			};
+		}),
+
+	getLessonData: authProcedure
```
Comment (spark-sse) [inline libs/data-access/api/src/lib/trpc/routers/lesson.router.ts:230]: again feels a little bit like "findOne" method. Check if you can reuse it

**Comment by AsadullahBajwa** [comment]
I tried using z.union, but it resulted in this error (see attached screenshot).

image

After researching, I found that the issue occurs because trpc-openapi requires the input parser to be a ZodObject and does not currently support ZodUnion, which causes the OpenAPI generation to fail.

So, I decided to go with the second approach by keeping both functions separate and following the suggested naming convention.

**Review by spark-sse — DISMISSED** [review]

**Comment by spark-sse** [comment]
can't merge due to the compiler error
